rest/test/server/bootstrapper_spec.js

```diff
@@ -751,6 +754,41 @@ describe('server (bootstrapper)', () => {
 		});
 	});
 
+	describe('HTTPS', () => {
+		it('creates https server with certificate and key given', done => {
+			createServer({
+				port: 3001,
+				protocol: 'HTTPS',
+				sslKeyPath: `${__dirname}/certs/restSSL.key`,
+				sslCertificatePath: `${__dirname}/certs/restSSL.crt`
+			});
+			done();
+		});
+
+		it('throws error when the key path is missing', done => {
+			expect(() => createServer({ port: 3001, protocol: 'HTTPS', sslCertificatePath: `${__dirname}/certs/restSSL.crt` }))
+				.to.throw('Server default is HTTPS but no SSL Key found, \'sslKeyPath\' property in the configuration must be provided.');
+			done();
+		});
+
+		it('throws error when the certificate path is missing', done => {
+			expect(() => createServer({ port: 3001, protocol: 'HTTPS', sslKeyPath: `${__dirname}/certs/restSSL.key` }))
+				.to.throw('Server default is HTTPS but no SSL Certificate found, '
+				+ '\'sslCertificatePath\' property in the configuration must be provided.');
+			done();
+		});
+
+		it('starts https and throws error when the protocol is not defined', done => {
+			expect(() => createServer({ port: 3001 })).to.throw();
+			done();
+		});
+
+		it('starts http when the protocol is HTTP', done => {
+			createServer({ port: 3001, protocol: 'HTTP' });
+			done();
+		});
+	});
```

Comment:
are you missing a success test?

Reply:
?

Reply:
Yes if I understood correctly; I've tried to have a HTTPS server routes test(s) earlier so that I can verify it's responding expectedly when on HTTPS; I couldn't find a way to overcome the "self-signed" certificate issue, this was one of the reasons the PR was in draft for now. But now I'm working on finding a way to overcome that issue.

Reply:
I found a workaround for hippie(a request wrapper used in the project) to call an HTTPS endpoint(doesn't work with hippie(httpsApp) unfortunately), created a test case to make sure it's responding expectedly, pushed, please take a look, it'd be nice to have a test case for `wss` as well.